torch/export/_trace.py

```diff
@@ -1369,66 +1416,38 @@ def forward(self, *args, **kwargs):
     range_constraints = make_constraints(
-        dynamo_fake_mode,
+        fake_mode,
         gm,
         combined_args,
         dynamic_shapes,
         num_lifted,
     )
+    if strict:
+        _add_runtime_assertions_to_cond_in_subgraph(
+            range_constraints,
+            gm,
+            fake_mode,
+        )
 
-    # Do some cleanups on the graph module to restore the state dict to the
-    # expected form. Each of these steps should probably get fixed upstream.
-    # 1. Remove tensor constants that were added as buffers.
-    _rewrite_dynamo_tensor_constants(
-        orig_mod_buffers=set(mod.buffers()),
-        traced_mod_buffers=dict(gm_torch_level.named_buffers()),
-        graph_signature=aten_export_artifact.sig,
-        constants=aten_export_artifact.constants,
-    )
-    # 2. Restore FQN of param/buffers
-    param_buffer_table: Dict[str, str] = _get_param_buffer_mapping(mod, gm_torch_level)
-    _replace_param_buffer_names(param_buffer_table, export_graph_signature)
-
-    # 3. Remove non-persistent buffers from the graph signature
-    _rewrite_non_persistent_buffers(
-        mod, aten_export_artifact.sig, aten_export_artifact.constants
-    )
-
-    # 4. Rewrite constants to have the same FQN as the original module.
-    _remap_constants(constant_attrs, export_graph_signature, constants)
-
-    # 5. Rename constants nodes in graph module from buffers to constants
-    _rename_constants_nodes(gm, export_graph_signature)
-
+    # Make module signatures.
     module_call_signatures = {
         fqn: ModuleCallSignature(inputs=[], outputs=[], **specs)
-        for fqn, specs in gm_torch_level.meta["module_call_specs"].items()
+        for fqn, specs in module_call_specs.items()
     }
 
     if len(preserve_module_call_signature) > 0:
+        if not strict:
+            _rewrite_node(gm)
         res = CollectTracepointsPass(module_call_signatures, export_graph_signature)(gm)
         assert res is not None
         gm = res.graph_module
 
-    # We can't get rid of this yet, since for some reason
-    # insert_deferred_runtime_assertions doesn't add assertions to cond
-    # subgraphs
-    if len(range_constraints) > 0:
-        stack_trace = (
-            'File "torch/_export/passes/add_runtime_assertions_for_constraints_pass.py", line 46, '
-            "in _AddRuntimeAssertionsForInlineConstraintsPass"
-        )
-        with dynamo_fake_mode, gm._set_create_node_hook(
-            functools.partial(_node_metadata_hook, stack_trace=stack_trace)
-        ):
-            res = _AddRuntimeAssertionsForInlineConstraintsPass(range_constraints)(gm)
-        assert res is not None
-        gm = res.graph_module
+    assert out_spec is not None
 
-    assert orig_out_spec is not None
     _verify_nn_module_stack(gm)
     _verify_stack_trace(gm)
-    _verify_placeholder_names(gm, export_graph_signature)
+    if strict or not _is_torch_jit_trace:
+        _verify_placeholder_names(gm, export_graph_signature)
```

Comment:
I'm a little confused by this casing -- the `if strict` part implies it shouldn't work for non-strict, but then it seems like the `not _is_torch_jit_trace` would cause this check to be run for non-strict export.

Reply:
Just tracing to the original code (https://github.com/pytorch/pytorch/blob/99af1b3ab03289e922b95252bab51758de6035c9/torch/export/_trace.py#L1208), I think this function happens for both strict and non-strict mode except when it runs on TS traced module. Does this make sense?

Reply:
Actually, as you mentioned, maybe we should just do `if not _is_torch_jit_trace:` :)